core/kt-osrd-rjs-parser/src/main/kotlin/fr/sncf/osrd/RawInfraRJSParser.kt

```diff
@@ -0,0 +1,788 @@
+                    if (
+                        previousElectrification.value != electrification.voltage &&
+                            previousElectrification.value != ""
+                    ) {
+                        logger.warn(
+                            "Electrification conflict on track-range ${electrificationRange.trackSectionID}" +
+                                "[${previousElectrification.lower + chunk.offset.distance}, " +
+                                "${previousElectrification.upper + chunk.offset.distance}]: " +
+                                "${previousElectrification.value} != ${electrification.voltage}"
+                        )
+                    }
+                }
+
+                chunk.electrificationVoltage.put(chunkLower, chunkUpper, electrification.voltage)
+            }
+        builder.applyFunctionToTrackSectionChunksBetween(
+            electrificationRange.trackSectionID,
+            electrificationRange.begin.meters,
+            electrificationRange.end.meters,
+            applyElectrificationForChunkBetween
+        )
+    }
+}
+
+fun parseNeutralRanges(
+    builder: RawInfraFromRjsBuilder,
+    isAnnouncement: Boolean,
+    neutralSection: RJSNeutralSection,
+) {
+    val trackRanges =
+        if (isAnnouncement) neutralSection.announcementTrackRanges else neutralSection.trackRanges
+    for (trackRange in trackRanges) {
+        val applyNeutralSectionForChunkBetween =
+            { chunk: TrackChunkDescriptor, chunkLower: Distance, chunkUpper: Distance ->
+                val track = trackRange.trackSectionID
+                val dir = trackRange.direction.toDirection()
+                val incomingNeutralSection =
+                    NeutralSection(neutralSection.lowerPantograph, isAnnouncement)
+                val dirChunkLower =
+                    if (dir == INCREASING) chunkLower else chunk.length.distance - chunkUpper
+                val dirChunkUpper =
+                    if (dir == INCREASING) chunkUpper else chunk.length.distance - chunkLower
+                val chunkDirNeutralSections = chunk.neutralSections.get(dir)
+
+                val previousDirNeutralSections =
+                    chunkDirNeutralSections.subMap(dirChunkLower, dirChunkUpper)
+                for (previousNeutralSection in previousDirNeutralSections) {
+                    if (previousNeutralSection.value != incomingNeutralSection) {
+                        logger.warn(
+                            "Neutral-section conflict on track-range $track.$dir" +
+                                "[$chunkLower, $chunkUpper]: " +
+                                "${previousNeutralSection.value} != $incomingNeutralSection"
+                        )
+                        break
+                    }
+                }
+
+                chunkDirNeutralSections.put(dirChunkLower, dirChunkUpper, incomingNeutralSection)
+
+                // TODO: remove this priority given to announcement.
```

Comment:
We have to remove this, too.

Reply:
I suggest we do it in a separate PR, to ensure this one changes as little behavior as possible